### PR TITLE
add velocity and acceleration to initial guess

### PR DESCRIPTION
**Pull request by Lewis-Seiden**

### CODE REVIEW

File: rust/examples/swerve.rs [hunk 1]
```diff
@@ -41,10 +41,11 @@ fn main() {
...
-    path.wpt_linear_velocity_polar(0, 0.0, 0.0);
-    path.wpt_linear_velocity_polar(1, 0.0, 0.0);
-    path.wpt_angular_velocity(0, 0.0);
-    path.wpt_angular_velocity(1, 0.0);
-    path.sgmt_circle_obstacle(0, 1, 0.5, 0.0, 0.2);
+    // path.wpt_linear_velocity_polar(0, 0.0, 0.0);
+    // path.wpt_linear_velocity_polar(1, 0.0, 0.0);
+    // path.wpt_angular_velocity(0, 0.0);
+    // path.wpt_angular_velocity(1, 0.0);
+    // path.sgmt_circle_obstacle(0, 1, 0.5, 0.1, 0.2);
```
Comment (calcmogul) [inline rust/examples/swerve.rs:48]: Commented out code should be removed. The example segfaulted though, so maybe these lines need to be uncommented instead.

**Review by calcmogul — CHANGES_REQUESTED** [review]

File: rust/examples/swerve.rs (hunk 1 above)
Comment (Lewis-Seiden) [reply rust/examples/swerve.rs:48]: the example worked on my machine, the segfault only happened when i tried to run it with the wrong flags

**Comment by calcmogul** [comment]
Here's a backtrace of the segfault:
```
Reading symbols from ./Swerve...
(gdb) run
Starting program: /home/tav/git/TrajoptLib/build-casadi/Swerve 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".

Program received signal SIGSEGV, Segmentation fault.
0x00007ffff7f80d95 in trajopt::ApplyInitialGuess<casadi::MX, trajopt::CasADiOpti> (opti=..., solution=..., 
    x=std::vector of length 5, capacity 5 = {...}, y=std::vector of length 5, capacity 5 = {...}, 
    theta=std::vector of length 5, capacity 5 = {...}, vx=std::vector of length 5, capacity 5 = {...}, 
    vy=std::vector of length 5, capacity 5 = {...}, omega=std::vector of length 5, capacity 5 = {...}, 
    ax=std::vector of length 5, capacity 5 = {...}, ay=std::vector of length 5, capacity 5 = {...}, 
    alpha=std::vector of length 5, capacity 5 = {...})
    at /home/tav/git/TrajoptLib/src/optimization/TrajoptUtil.inc:361
361         opti.SetInitial(vx[sampleIndex],
(gdb) bt
#0  0x00007ffff7f80d95 in trajopt::ApplyInitialGuess<casadi::MX, trajopt::CasADiOpti> (opti=..., 
    solution=..., x=std::vector of length 5, capacity 5 = {...}, 
    y=std::vector of length 5, capacity 5 = {...}, theta=std::vector of length 5, capacity 5 = {...}, 
    vx=std::vector of length 5, capacity 5 = {...}, vy=std::vector of length 5, capacity 5 = {...}, 
    omega=std::vector of length 5, capacity 5 = {...}, ax=std::vector of length 5, capacity 5 = {...}, 
    ay=std::vector of length 5, capacity 5 = {...}, alpha=std::vector of length 5, capacity 5 = {...})
    at /home/tav/git/TrajoptLib/src/optimization/TrajoptUtil.inc:361
#1  0x00007ffff7f86db1 in trajopt::SwerveDiscreteOptimal<casadi::MX, trajopt::CasADiOpti>::SwerveDiscreteOptimal (this=0x7fffffffd6a0, path=..., N=std::vector of length 1, capacity 1 = {...}, initialGuess=...)
    at /home/tav/git/TrajoptLib/src/optimization/algorithms/SwerveDiscreteOptimal.inc:148
#2  0x00007ffff7f7ec4a in trajopt::OptimalTrajectoryGenerator::Generate (path=...)
    at /home/tav/git/TrajoptLib/src/OptimalTrajectoryGenerator.cpp:26
#3  0x00005555555577af in main () at /home/tav/git/TrajoptLib/examples/Swerve/src/Main.cpp:46
(gdb) p sampleIndex
$1 = 1
```

**Comment by calcmogul** [comment]
It segfaults because solution.dt is empty (solution.dt.size() == 0).

**Review by calcmogul — APPROVED** [review]

**Pull request merged**